GVRf/Framework/jni/objects/components/camera_rig_jni.cpp

```diff
@@ -197,16 +195,22 @@ Java_org_gearvrf_NativeCameraRig_setCameraSeparationDistance(
     camera_rig->set_camera_separation_distance(distance);
 }
 
-JNIEXPORT jfloat JNICALL
+JNIEXPORT jfloatArray JNICALL
 Java_org_gearvrf_NativeCameraRig_getFloat(JNIEnv * env,
         jobject obj, jlong jcamera_rig, jstring key) {
     std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
             CameraRig>*>(jcamera_rig);
     const char* char_key = env->GetStringUTFChars(key, 0);
     std::string native_key = std::string(char_key);
-    jfloat camera_rig_float = camera_rig->getFloat(native_key);
     env->ReleaseStringUTFChars(key, char_key);
-    return camera_rig_float;
+    try {
+        jfloat value = camera_rig->getFloat(native_key);
+        jfloatArray result_float = env->NewFloatArray(1);
+        env->SetFloatArrayRegion(result_float, 0, 1, &value);
+        return result_float;
+    } catch (const std::string & err) {
+        return NULL;
```

Comment:
Jason, you’ve got two programmers telling you that they disagree. NULL is just as much a special pointer value as NaN is a special float value … and creating an array adds cost.

From: jason2kim [mailto:notifications@github.com]
Sent: Wednesday, May 06, 2015 12:14 PM
To: Samsung/GearVRf
Cc: Jon Shemitz
Subject: Re: [GearVRf] fix for some similar issues of, (#31)


In GVRf/Framework/jni/objects/components/camera_rig_jni.cpp<https://github.com/Samsung/GearVRf/pull/31#discussion_r29791698>:

>      env->ReleaseStringUTFChars(key, char_key);

> -    return camera_rig_float;

> +    try {

> +        jfloat value = camera_rig->getFloat(native_key);

> +        jfloatArray result_float = env->NewFloatArray(1);

> +        env->SetFloatArrayRegion(result_float, 0, 1, &value);

> +        return result_float;

> +    } catch (const std::string & err) {

> +        return NULL;

More correct way would be using a separate parameter as error code and another one as return value, like float and vec arrary. But, considering JNI, I think the current implementation is the optimal, with minimum confusion.

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/31/files#r29791698>.